src/discof/repair/fd_repair_tile.c

```diff
@@ -434,6 +688,266 @@ during_frag( fd_repair_tile_ctx_t * ctx,
+        glob->metrics.recv_serv_pkt_types[FD_METRICS_ENUM_REPAIR_SERV_PKT_TYPES_V_PONG_IDX]++;
+        fd_repair_recv_pong( glob, &protocol->inner.pong, peer_addr );
+
+        return 0;
+      case fd_repair_protocol_enum_window_index: {
+        glob->metrics.recv_serv_pkt_types[FD_METRICS_ENUM_REPAIR_SERV_PKT_TYPES_V_WINDOW_IDX]++;
+        fd_repair_window_index_t * wi = &protocol->inner.window_index;
+        header = &wi->header;
+        break;
+      }
+      case fd_repair_protocol_enum_highest_window_index: {
+        glob->metrics.recv_serv_pkt_types[FD_METRICS_ENUM_REPAIR_SERV_PKT_TYPES_V_HIGHEST_WINDOW_IDX]++;
+        fd_repair_highest_window_index_t * wi = &protocol->inner.highest_window_index;
+        header = &wi->header;
+        break;
+      }
+      case fd_repair_protocol_enum_orphan: {
+        glob->metrics.recv_serv_pkt_types[FD_METRICS_ENUM_REPAIR_SERV_PKT_TYPES_V_ORPHAN_IDX]++;
+        fd_repair_orphan_t * wi = &protocol->inner.orphan;
+        header = &wi->header;
+        break;
+      }
+      default: {
+        glob->metrics.recv_serv_pkt_types[FD_METRICS_ENUM_REPAIR_SERV_PKT_TYPES_V_UNKNOWN_IDX]++;
+        FD_LOG_WARNING(( "received repair request of unknown type: %d", (int)protocol->discriminant ));
+        return 0;
+      }
+    }
+
+    if( FD_UNLIKELY( !fd_hash_eq( &header->recipient, glob->public_key ) ) ) {
+      FD_LOG_WARNING(( "received repair request with wrong recipient, %s instead of %s", FD_BASE58_ENC_32_ALLOCA( header->recipient.uc ), FD_BASE58_ENC_32_ALLOCA( glob->public_key ) ));
+      return 0;
+    }
+
+    /* Verify the signature */
+    fd_sha512_t sha2[1];
+    fd_signature_t sig;
+    fd_memcpy( &sig, header->signature.uc, sizeof(sig) );
+    fd_memcpy( (uchar *)msg + 64U, msg, 4U );
+    if( fd_ed25519_verify( /* msg */ msg + 64U,
+                           /* sz */ msglen - 64U,
+                           /* sig */ sig.uc,
+                           /* public_key */ header->sender.uc,
+                           sha2 )) {
+      glob->metrics.recv_serv_invalid_signature++;
+      FD_LOG_WARNING(( "received repair request with with invalid signature" ));
+      return 0;
+    }
+
+    fd_pinged_elem_t * val = fd_pinged_table_query( glob->pinged, peer_addr, NULL) ;
+    if( val == NULL || !val->good || !fd_hash_eq( &val->id, &header->sender ) ) {
+      /* Need to ping this client */
+      if( val == NULL ) {
+        if( fd_pinged_table_is_full( glob->pinged ) ) {
+          FD_LOG_WARNING(( "pinged table is full" ));
+
+          glob->metrics.recv_serv_full_ping_table++;
+          return 0;
+        }
+        val = fd_pinged_table_insert( glob->pinged, peer_addr );
+        for ( ulong i = 0; i < FD_HASH_FOOTPRINT / sizeof(ulong); i++ )
+          val->token.ul[i] = fd_rng_ulong(glob->rng);
+      }
+      fd_hash_copy( &val->id, &header->sender );
+      val->good = 0;
+      fd_repair_send_ping( repair_tile_ctx, glob, peer_addr, self_ip4_addr, val );
```

Comment:
same, but dont worry about changing this function since we don't need it right now